### PR TITLE
feat: adds a Flask CRUD template for Python Lambda Functions

### PR DESCRIPTION
**Pull request by aspittel**
*Description of changes:*

Adds a Flask CRUD template for Python Lambda Functions. This is far from perfect, make/request any changes needed, but it gets closer to the CRUD template given for Node.js functions.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by reloadedhead** [comment]
Having read this commit and `awsgi` repository, I have a doubt: how does awsgi handle the Event object into the request  object?

**Comment by aspittel** [comment]
Here's the relevant code @reloadedhead https://github.com/slank/awsgi/blob/2cf2ec2995683dfa72a0722abde3ecfdabeceb41/awsgi/__init__.py#L112

**Review by dabit3 — DISMISSED** [review]

**Comment by dabit3** [comment]
This looks awesome @aspittel, goes towards some of the goals around REST API improvements this year for sure.

@renebrandel, what is the timeline for getting this merged?

Thanks,

**Comment by attilah** [comment]
@dabit3 we need to check if the e2e tests are passing before merging it, perhaps they need an update.

File: packages/amplify-python-function-template-provider/amplify-plugin.json [hunk 1]
```diff
@@ -13,6 +13,10 @@
       {
         "name": "Hello World",
         "value": "hello-world"
+      },
```
Comment (edwardfoyle) [inline packages/amplify-python-function-template-provider/amplify-plugin.json:16]: python e2e tests will need an update to "know about" this template option in the list

File: packages/amplify-python-function-template-provider/resources/crud/event.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "test": "event" }
```
Comment (edwardfoyle) [inline packages/amplify-python-function-template-provider/resources/crud/event.json:1]: does this event work out of the box with the function? Seems like I remember an event like `{ "path": "/items", "httpMethod": "GET" }` was necessary for it to work

Comment (aspittel) [reply packages/amplify-python-function-template-provider/resources/crud/event.json:1]: You're so right. Thank you for catching that! Updating now.

**Comment by danielleadams** [comment]
Hey @aspittel! Are you still interested in getting this merged in?

**Comment by danielleadams** [comment]
Closing due to inactivity

**Pull request closed without merging**